src/components/nc-slideshow.vue

```diff
@@ -0,0 +1,241 @@
+          :key=index @click.stop="selectNewSlide(index)" 
+          :class="[ slideIndex === index ? paginationActiveClass : '']" 
+          role="tab"
+          :aria-label="ariaObject.itemIndex + (index + 1)"
+          :aria-controls="ariaObject.tabPanel+(index + 1)"
+          :aria-selected="slideIndex === index ? true : false"
+          :tabindex="slideIndex === index ? -1 : 0"
+        >
+          <span>{{ ariaText.ariaTextDots + index + 1 }}</span>
+        </li>
+      </ul>
+    </div>
+    <a 
+      v-if="hasLinkLeft" 
+      class="nc-slideshow__link--left" 
+      @click="leftLinkHandler($event)" 
+      :data-slide-to="slideIndex + 1"
+      :style="leftLinkStyle"
+      role="button"
+    >
+      {{ leftLinkText}}
+    </a>
+    <a 
+      v-if="hasLinkRight" 
+      class="nc-slideshow__link--right"
+      @click="rightLinkHandler($event)" 
+      :data-slide-to="slideIndex + 1"
+      :style="rightLinkStyle"
+      role="button"
+    >
+      {{ rightLinkText }}
+    </a>
+  </div>
+  
+</template>
+
+<script>
+/**
+ * Code based on https://github.com/myliang/fish-ui/blob/master/src/components/Carousel.vue
+ */
+export default {
+  name: 'nc-slideshow',
+  props: {
+    paginationActiveClass: {
+      type: String,
+      default: 'active'
+    },
+    paginationStyle: {
+      type: Object
+    },
+    hasLinkLeft: {
+      type: Boolean,
+      default: false
+    },
+    hasLinkRight: {
+      type: Boolean,
+      default: false
+    },
+    leftLinkText: {
+      type: String,
+      default: 'Previous'
+    },
+    rightLinkText: {
+      type: String,
+      default: 'Next'
+    },
+    linksDefaultAction: {
+      type: Boolean,
+      default: false
+    },
+    leftLinkStyle: {
+      type: Object
+    },
+    rightLinkStyle: {
+      type: Object
+    },
+    ariaText: {
+      type: Object,
+      default: function() {
+        return {
+          ariaTextDots: 'Slide'
+        }
+      }
+    }
+  },
+  data() {
+    return {
+      slideIndex: 0,
+      slideLength: 0,
+      width: 0,
+      ariaObject: {
+        itemIndex: 'slide',
+        tabPanel: 'tabpanel-0-'
+      }
+    }
+  },
+  methods: {
+    selectNewSlide(index) {
+      this.slideIndex = index
+    },
+    leftLinkHandler(e) {
+      if (this.linksDefaultAction) {
+        this.goToSlide('right')
+      }
+      if (this.slideIndex === 0) {
+        this.$emit('slideshow-first-slide', e)
+      } else {
+        this.$emit('slideshow-click-left-link', e)
+      }
+    },
+    rightLinkHandler(e) {
+      if (this.linksDefaultAction) {
+        this.goToSlide('left')
+      }
+      if (this.slideIndex === this.slideLength - 1) {
+        this.$emit('slideshow-last-slide', e)
+      } else {
+        this.$emit('slideshow-click-right-link', e)
+      }
+    },
+    goToSlide(direction) {
+      if (direction === 'left') {
+        this.slideIndex < this.slideLength - 1
+          ? ++this.slideIndex
+          : this.slideIndex
```

Comment:
```suggestion
                this.slideIndex < this.slideLength - 1 ? ++this.slideIndex : this.slideIndex
```
Esto es una sugerencia, no mas, lo digo porque normalmente los operadores ternarios los veo en una línea, y no parece que sea muy larga. Pero no obstante que @rafagarcia  o @AlbertoFdzM  dictaminen.

Reply:
Esto lo cambia el lint y lo pone en dos líneas, el problema es la longitud de la línea, si no sí estaría en una sola.

Reply:
ok =)

Reply:
Yo utilizaría operadores ternarios sólo para asignación a variables, aquí se está usando para una ejecución `++this.slideIndex`. Lo encuentro dificil de leer ya que me ha costado entender que la otra parte de este "if" reducido no hace nada.

yo me generaría una vairbla helper `nextSlideIndex` para ayudar a la lectura y pasados todos estos ifs la asignaria al `this.slideIndex`

Reply:
Modificado